sqlg-core/src/main/java/org/umlg/sqlg/structure/topology/Index.java

```diff
@@ -29,6 +29,32 @@
     private List<PropertyColumn> properties = new ArrayList<>();
     private IndexType uncommittedIndexType;
     private List<PropertyColumn> uncommittedProperties = new ArrayList<>();
+    private PartialIndexClause partialIndexClause = null;
```

Comment:
It took me some time to understand the diff between the change in Index.java versus the change in EdgeLabel.java.
So with the change in EdgeLabel, we create the partial index (i.e., NOT NULL) when EdgeTable is created with a flag? On the other hand, the change in index.java here is a more general way to define a PartialIndexClause - we need to use AbstractLabel.ensureIndexExists to actually apply it?

Probably nice to add some inline comment or changelog (e.g., concrete PartialIndexClause example) to demonstrate that.

Reply:
That is a good suggestion. I will add some inline comments to PartialIndexClause, since changelog will not provide enough space and context to explain it.

Reply:
Here is the comment that I added.
```
/**  PartialIndexClause allows to add any conditional clause for a property column while adding an index to a 
 * Vertex or Edge. This is used with ensureIndexExists method which is used to add user defined additional index to
 * a Vertex or an Edge.  *  * For implicit index created during Edge creation itself, use the boolean argument
 * usePartialIndexForForeignKeyNotNull with EdgeLabel#createEdgeLabel. 
 */
```